server/src/main/java/org/opensearch/index/compositeindex/datacube/NumericDimension.java

```diff
@@ -33,6 +35,22 @@ public String getField() {
         return field;
     }
 
+    @Override
+    public int getNumSubDimensions() {
+        return 1;
```

Comment:
so every dimension, has a subdimension, which is itself?

Reply:
sub dimension can be one ( the actual dimension value ) or many ( like in date - calendar intervals are the sub dimensions )